File: ruby/lib/ik_classifications/absolute_size_classification.rb
# A test that can rule out some graphs from being IK.  This is based
# on Robertson, Seymour, and Thomas' work.
class AbsoluteSizeClassification < IKClassification::Base
  def classify(graph)
    graph.size < 15 ? IS_NOT_IK : CANNOT_DETERMINE_IK
  end
  
  def description
    "If there are less than 15 edges then the graph is " +
    "NOT intrinsically knotted."
  end
end

File: java/src/ik/AbsoluteSizeClassification.java
package ik;

/** 
 * A test that can rule out some graphs from being IK.  This is based
 * on Robertson, Seymour, and Thomas’ work.
 */
public class AbsoluteSizeClassification implements IKClassification {
  public String classify(Graph graph) {  
    return (graph.getSize() < 15) ? IS_NOT_IK : CANNOT_DETERMINE_IK;
  }

  public String getName() {
    return "AbsoluteSizeClassification";  
  }
  
  public String getDescription() {
    return "If there are less than 15 edges then the graph is " +
           "NOT intrinsically knotted.";
  }
}